failure on ud_*/test_uct_peer_failure_multiple.test

```
[2020-05-12T12:23:09.101Z] [ RUN      ] ud_verbs/test_uct_peer_failure_multiple.test/0 <ud_verbs/mlx4_0:1>
[2020-05-12T12:23:17.191Z] [boo31:6130 :1:11538]       ud_ep.c:1034 Assertion `UCT_UD_PSN_COMPARE(ep->resend.max_psn, >, ep->tx.acked_psn)' failed
[2020-05-12T12:23:17.191Z] 
[2020-05-12T12:23:17.191Z] /scrap/jenkins/workspace/ucx-3/contrib/../src/uct/ib/ud/base/ud_ep.c: [ uct_ud_ep_resend() ]
[2020-05-12T12:23:17.191Z]       ...
[2020-05-12T12:23:17.191Z]      1031     uint16_t iovcnt;
[2020-05-12T12:23:17.191Z]      1032 
[2020-05-12T12:23:17.191Z]      1033     ucs_assert_always(UCT_UD_PSN_COMPARE(ep->resend.max_psn, >, ep->tx.acked_psn));
[2020-05-12T12:23:17.191Z] ==>  1034 
[2020-05-12T12:23:17.191Z]      1035     /* check window */
[2020-05-12T12:23:17.191Z]      1036     resend_pos = ep->resend.pos;
[2020-05-12T12:23:17.191Z]      1037     if (ucs_queue_iter_end(&ep->tx.window, resend_pos)) {
[2020-05-12T12:23:17.191Z] 
[2020-05-12T12:23:17.191Z] ==== backtrace (tid:  11538) ====
[2020-05-12T12:23:17.191Z]  0 0x0000000000027edb uct_ud_ep_resend()  /scrap/jenkins/workspace/ucx-3/contrib/../src/uct/ib/ud/base/ud_ep.c:1034
[2020-05-12T12:23:17.191Z]  1 0x000000000002973b uct_ud_ep_do_pending()  /scrap/jenkins/workspace/ucx-3/contrib/../src/uct/ib/ud/base/ud_ep.c:1302
[2020-05-12T12:23:17.191Z]  2 0x00000000000399c9 ucs_arbiter_dispatch_nonempty()  /scrap/jenkins/workspace/ucx-3/contrib/../src/ucs/datastruct/arbiter.c:321
[2020-05-12T12:23:17.191Z]  3 0x000000000002e494 uct_ud_iface_progress_pending()  /scrap/jenkins/workspace/ucx-3/contrib/../src/uct/ib/ud/base/ud_iface.h:421
[2020-05-12T12:23:17.191Z]  4 0x000000000002e494 uct_ud_verbs_iface_async_progress()  /scrap/jenkins/workspace/ucx-3/contrib/../src/uct/ib/ud/verbs/ud_verbs.c:410
[2020-05-12T12:23:17.191Z]  5 0x0000000000022da9 uct_ud_iface_async_progress()  /scrap/jenkins/workspace/ucx-3/contrib/../src/uct/ib/ud/base/ud_iface.c:813
[2020-05-12T12:23:17.191Z]  6 0x000000000002357c uct_ud_iface_timer()  /scrap/jenkins/workspace/ucx-3/contrib/../src/uct/ib/ud/base/ud_iface.c:824
[2020-05-12T12:23:17.191Z]  7 0x000000000002357c uct_ud_iface_timer()  /scrap/jenkins/workspace/ucx-3/contrib/../src/uct/ib/ud/base/ud_iface.c:825
[2020-05-12T12:23:17.191Z]  8 0x00000000000311f2 ucs_async_handler_invoke()  /scrap/jenkins/workspace/ucx-3/contrib/../src/ucs/async/async.c:224
[2020-05-12T12:23:17.191Z]  9 0x00000000000312e5 ucs_async_handler_dispatch()  /scrap/jenkins/workspace/ucx-3/contrib/../src/ucs/async/async.c:244
[2020-05-12T12:23:17.191Z] 10 0x00000000000318a8 ucs_async_dispatch_handlers()  /scrap/jenkins/workspace/ucx-3/contrib/../src/ucs/async/async.c:273
[2020-05-12T12:23:17.191Z] 11 0x0000000000031c8f ucs_async_dispatch_timerq()  /scrap/jenkins/workspace/ucx-3/contrib/../src/ucs/async/async.c:300
[2020-05-12T12:23:17.191Z] 12 0x0000000000035332 ucs_async_thread_func()  /scrap/jenkins/workspace/ucx-3/contrib/../src/ucs/async/thread.c:138
[2020-05-12T12:23:17.191Z] 13 0x0000003479c07aa1 start_thread()  pthread_create.c:0
[2020-05-12T12:23:17.191Z] 14 0x00000034798e8bcd __clone()  ???:0
[2020-05-12T12:23:17.191Z] =================================
[2020-05-12T12:23:17.191Z] Sending notification to alinas@mellanox.com
[2020-05-12T12:23:17.191Z] [boo31:6130 :1:11538] Process frozen...
```

http://hpc-master.lab.mtl.com:8080/blue/organizations/jenkins/ucx/detail/ucx/4877/pipeline/


```
[ RUN      ] ud_mlx5/test_uct_peer_failure_multiple.test/0 <ud_mlx5/mlx5_0:1>
[swx-rdmz-ucx-new-01:26270:1:17983]       ud_ep.c:1034 Assertion `UCT_UD_PSN_COMPARE(ep->resend.max_psn, >, ep->tx.acked_psn)' failed

/scrap/azure/agent-02/AZP_WORKSPACE/2/s/contrib/../src/uct/ib/ud/base/ud_ep.c: [ uct_ud_ep_resend() ]
      ...
     1031     uint16_t iovcnt;
     1032 
     1033     ucs_assert_always(UCT_UD_PSN_COMPARE(ep->resend.max_psn, >, ep->tx.acked_psn));
==>  1034 
     1035     /* check window */
     1036     resend_pos = ep->resend.pos;
     1037     if (ucs_queue_iter_end(&ep->tx.window, resend_pos)) {

==== backtrace (tid:  17983) ====
 0 0x00000000000595a8 ucs_debug_print_backtrace()  /scrap/azure/agent-02/AZP_WORKSPACE/2/s/contrib/../src/ucs/debug/debug.c:656
 1 0x0000000000071a0b uct_ud_ep_resend()  /scrap/azure/agent-02/AZP_WORKSPACE/2/s/contrib/../src/uct/ib/ud/base/ud_ep.c:1034
 2 0x0000000000071a0b uct_ud_ep_prepare_crep()  /scrap/azure/agent-02/AZP_WORKSPACE/2/s/contrib/../src/uct/ib/ud/base/ud_ep.c:992
 3 0x0000000000071a0b uct_ud_ep_do_pending_ctl()  /scrap/azure/agent-02/AZP_WORKSPACE/2/s/contrib/../src/uct/ib/ud/base/ud_ep.c:1212
 4 0x0000000000073574 uct_ud_ep_do_pending()  /scrap/azure/agent-02/AZP_WORKSPACE/2/s/contrib/../src/uct/ib/ud/base/ud_ep.c:1302
 5 0x000000000004edd0 ucs_arbiter_dispatch_nonempty()  /scrap/azure/agent-02/AZP_WORKSPACE/2/s/contrib/../src/ucs/datastruct/arbiter.c:321
 6 0x000000000007b057 ucs_arbiter_dispatch()  /scrap/azure/agent-02/AZP_WORKSPACE/2/s/contrib/../src/ucs/datastruct/arbiter.h:380
 7 0x000000000007b057 uct_ud_iface_progress_pending()  /scrap/azure/agent-02/AZP_WORKSPACE/2/s/contrib/../src/uct/ib/ud/base/ud_iface.h:421
 8 0x000000000007b057 uct_ud_mlx5_iface_async_progress()  /scrap/azure/agent-02/AZP_WORKSPACE/2/s/contrib/../src/uct/ib/ud/accel/ud_mlx5.c:542
 9 0x000000000006bb32 uct_ud_iface_async_progress()  /scrap/azure/agent-02/AZP_WORKSPACE/2/s/contrib/../src/uct/ib/ud/base/ud_iface.c:813
10 0x000000000006bb32 uct_ud_iface_timer()  /scrap/azure/agent-02/AZP_WORKSPACE/2/s/contrib/../src/uct/ib/ud/base/ud_iface.c:824
11 0x0000000000045807 ucs_async_handler_invoke()  /scrap/azure/agent-02/AZP_WORKSPACE/2/s/contrib/../src/ucs/async/async.c:224
12 0x0000000000045807 ucs_async_dispatch_handlers()  /scrap/azure/agent-02/AZP_WORKSPACE/2/s/contrib/../src/ucs/async/async.c:273
13 0x0000000000045c01 ucs_async_dispatch_timerq()  /scrap/azure/agent-02/AZP_WORKSPACE/2/s/contrib/../src/ucs/async/async.c:300
14 0x0000000000049885 ucs_async_thread_func()  /scrap/azure/agent-02/AZP_WORKSPACE/2/s/contrib/../src/ucs/async/thread.c:138
15 0x0000000000007dd5 start_thread()  pthread_create.c:0
16 0x00000000000fdead __clone()  ???:0
=================================
[swx-rdmz-ucx-new-01:26270:1:17983] Process frozen...
make: *** [test] Terminated

```

https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=6036&view=logs&j=2a24149b-54e0-5984-9c93-20e052df9aa1&t=6985259c-90af-5b9c-3e23-537f385bafac

--- a/src/uct/ib/ud/base/ud_ep.c
+++ b/src/uct/ib/ud/base/ud_ep.c
@@ -1031,7 +1031,11 @@ static void uct_ud_ep_resend(uct_ud_ep_t *ep)
     int max_log_sge;
     uint16_t iovcnt;
 
-    ucs_assert_always(UCT_UD_PSN_COMPARE(ep->resend.max_psn, >, ep->tx.acked_psn));
+    /* check if the resend window was acknowledged */
+    if (UCT_UD_PSN_COMPARE(ep->resend.max_psn, <=, ep->tx.acked_psn)) {
+        uct_ud_ep_ctl_op_del(ep, UCT_UD_EP_OP_RESEND);
+        return;
+    }
 
     /* check window */
     resend_pos = ep->resend.pos;

